app/scripts/services/opensensemapapi.js

```diff
@@ -92,12 +92,12 @@
         .catch(failed);
     }
 
-    function postMeasurements (boxId, measurements, format) {
+    function postMeasurements (boxId, measurements, format,access_token) {
       var url = getUrl() + '/boxes/' + boxId + '/data';
 
       return $http
         .post(url, measurements, {
-          headers: { 'content-type': format }
+          headers: { 'content-type': format ,'Authorization': access_token}
```

Comment:
Have you tested this for boxes which do not require authorization?

```suggestion
          headers: { 'content-type': format , 'Authorization': access_token}
```

Reply:
I did ! Everything was working as expected